runtime/lua/vim/filetype/detect.lua

```diff
@@ -45,6 +45,17 @@ function M.asm(bufnr)
   end
 end
 
+-- Active Server Pages (with Perl or Visual Basic Script)
+function M.asp(bufnr)
+  if vim.g.filetype_asp then
+    return vim.g.filetype_asp
+  elseif table.concat(getlines(bufnr, 1, 3)):lower():find('perlscript') then
```

Comment:
Would it make sense to add an optional argument (table) to the `getlines` function that allows passing `concat = true`?

Reply:
Ah, OK. Then I defer to him -- never ping pong ;) (Changes shouldn't cycle; always optimize over the full span of the previous solutions so they improve on each.)